drv/i2c-devices/src/lib.rs

```diff
@@ -75,6 +76,50 @@ macro_rules! pmbus_read {
     };
 }
 
+macro_rules! pmbus_rail_read {
+    ($device:expr, $rail:expr, $cmd:ident) => {{
+        let payload = [PAGE::CommandData::code(), $rail];
+
+        match $cmd::CommandData::from_slice(&match $device
+            .write_read_reg::<u8, [u8; $cmd::CommandData::len()]>(
+                $cmd::CommandData::code(),
+                &payload,
+            ) {
+            Ok(rval) => Ok(rval),
+            Err(code) => Err(Error::BadRead {
+                cmd: $cmd::CommandData::code(),
+                code,
+            }),
+        }?) {
+            Some(data) => Ok(data),
+            None => Err(Error::BadData {
+                cmd: $cmd::CommandData::code(),
+            }),
+        }
+    }};
+
+    ($device:expr, $rail:expr, $dev:ident::$cmd:ident) => {{
+        let payload = [$dev::PAGE::CommandData::code(), $rail];
+
+        match $dev::$cmd::CommandData::from_slice(&match $device
+            .write_read_reg::<u8, [u8; $dev::$cmd::CommandData::len()]>(
+                $dev::$cmd::CommandData::code(),
+                &payload,
+            ) {
+            Ok(rval) => Ok(rval),
+            Err(code) => Err(Error::BadRead {
+                cmd: $dev::$cmd::CommandData::code(),
+                code,
+            }),
+        }?) {
+            Some(data) => Ok(data),
+            None => Err(Error::BadData {
+                cmd: $dev::$cmd::CommandData::code(),
+            }),
+        }
```

Comment:
```suggestion
        use $dev::{$cmd, PAGE};
        pmbus_rail_read!($device, $rail, $cmd)
```
(see explanation below; this could apply to every `pmbus_xyz!` macro)

Reply:
Done -- thank you again!